apps/fhir/server/authentication.py

```diff
@@ -99,7 +103,8 @@ def search_fhir_id_by_identifier(search_identifier, request=None):
                 raise UpstreamServerException(err_detail)
             return fhir_id
         except requests.exceptions.RequestException as e:
-            if retries < max_retries:
+            if retries < max_retries and os.environ.get('TARGET_ENV') is None:
```

Comment:
I do see some references to `TARGET_ENV`, not sure exactly when/if that would come up, but we may want to try to check for that too, basically if TARGET_ENV is either None or "dev" then honor the retries. Probably optional 🤷‍♂️